assets/app/actions/siteActions.js

```diff
@@ -55,6 +55,23 @@ export default {
     }).catch(error => alertActions.httpError(error.message));
   },
 
+  uploadFile(site, path, fileData, message) {
```

Comment:
Why is the file data already base64 encoded? Is there a way that encoding or decoding only happens in actions and components just work with things like human readable strings and file objects?